### PR TITLE
Use a more illustrative example for tagged template literals

### LINKED ISSUE #13701: Issue with "Grammar and types": tagged template snippet unhelpful
MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Grammar_and_types

#### What information was incorrect, unhelpful, or incomplete?
I find the code snippet for tagged templates unhelpful.

#### Specific section or headline?
Under String Literals, see the 4 line code snippet beginning with "let myTag ="

#### What did you expect to see?
This particular example does not motivate the use of tagged templates. If you remove the first line and the tag expression from the third line, you get the same result:

```
let [name, age] = ['Mika', 28];
`Participant "${ name }" is ${ age } years old.`;
// Participant "Mika" is 28 years old.
```

Please use an example which explains why and when tagged templates are useful.

#### Did you test this? If so, how?

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/javascript/guide/grammar_and_types`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Grammar_and_types
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/guide/grammar_and_types/index.md
* Last commit: https://github.com/mdn/content/commit/426485595b1be50e2af91764fbfd94c04c3c36a5
* Document last modified: 2022-02-18T11:38:26.000Z

</details>

### PR DESCRIPTION
**Pull request by Josh-Cena**
#### Motivation

Fix #13701

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

**Comment by Josh-Cena** [comment]
Build failure?? Sus...

File: files/en-us/web/javascript/guide/grammar_and_types/index.md [hunk 1]
```diff
@@ -644,19 +644,70 @@ Template literals provide syntactic sugar for constructing strings. (This is sim
...
+  return arg;
+}
+
+const print = (segments, ...args) => {
+  // For any well-formed template literal, there will always be N args and
+  // (N+1) string segments.
+  let message = segments[0];
+  segments.slice(1).forEach((segment, index) => {
+    message += formatArg(args[index]) + segment;
+  });
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/guide/grammar_and_types/index.md:673]: Purposefully not use `segments.reduce` here, to avoid confusing learners.

**Review by teoli2003 — APPROVED** [review]
👍

File: files/en-us/web/javascript/guide/grammar_and_types/index.md [hunk 2]
```diff
@@ -644,19 +644,70 @@ Template literals provide syntactic sugar for constructing strings. (This is sim
...
+const todos = [
+  "Learn JavaScript",
+  "Learn Web APIs",
+  "Set up my website",
+  "Profit!",
+];
+
+print`I need to do:
+${todos}
+My current progress is: ${{ javascript: 20, html: 50, css: 10 }}
```
Comment (teoli2003) [inline files/en-us/web/javascript/guide/grammar_and_types/index.md:686]: I think this is where Yari is complaining. We may need to escape the first brace.

Comment (Josh-Cena) [reply files/en-us/web/javascript/guide/grammar_and_types/index.md:686]: Yari trying to parse code blocks seems like a bug to me. I might just extract the object to a variable for the time being.

**Pull request merged**